api2-war/src/main/java/eu/europeana/api2/v2/web/controller/ObjectController.java

```diff
@@ -380,25 +419,31 @@ private ModelAndView generateSchemaOrg(FullBean bean, RequestData data) {
     }
 
     private ModelAndView generateJsonLd(FullBean bean, RequestData data, HttpServletResponse response) {
-        String jsonld;
+        OutputStream outputStream = new ByteArrayOutputStream();
         String rdf    = EdmUtils.toEDM((FullBeanImpl) bean);
         try (InputStream rdfInput = IOUtils.toInputStream(rdf)) {
-            Model  modelResult = ModelFactory.createDefaultModel().read(rdfInput, "", "RDF/XML");
-            Object raw         = JSONLD.fromRDF(modelResult, new JenaRDFParser());
-            if (StringUtils.equalsIgnoreCase(data.profile, "compacted")) {
-                raw = JSONLD.compact(raw, jsonldContext, new Options());
-            } else if (StringUtils.equalsIgnoreCase(data.profile, "flattened")) {
-                raw = JSONLD.flatten(raw);
-            } else if (StringUtils.equalsIgnoreCase(data.profile, "normalized")) {
-                raw = JSONLD.normalize(raw);
+             Model  modelResult = ModelFactory.createDefaultModel().read(rdfInput, "", "RDF/XML");
+
+             if (StringUtils.equalsIgnoreCase(data.profile, "compacted")) {
+                 DatasetGraph graph = DatasetFactory.wrap(modelResult).asDatasetGraph();
+                 PrefixMap pm = RiotLib.prefixMap(graph);
+                 JsonLDWriteContext ctx = new JsonLDWriteContext();
+                 ctx.setJsonLDContext(ObjectController.jsonldContext);
+                 WriterDatasetRIOT writer = RDFDataMgr.createDatasetWriter(RDFFormat.JSONLD_FLATTEN_PRETTY);
+                 writer.write(outputStream, graph, pm, null, ctx);
+             }
+            else if (StringUtils.equalsIgnoreCase(data.profile, "flattened")) {
+                 RDFDataMgr.write(outputStream, modelResult, RDFFormat.JSONLD_FLAT);
+             } else if (StringUtils.equalsIgnoreCase(data.profile, "normalized")) {
+                //Did not found a normalized format for JSON-LD. Hence for now, I have used JSON-LD_PRETTY
```

Comment:
I discussed with Hugo, we can remove support for the different formats (see also his latest comment on the ticket).
I propose to return a one-line string, just like we do for json (most efficient). Users can rely on browser plugins to change this into something that is better readable.

Reply:
Have removed it completely for json-ld. Also have kept the format **JSONLD_FLATTEN_PRETTY** which results in the default format output as we get now currently for json-ld

Reply:
Right now json-ld requests return all data in 1 line. If you see any formatting then this is because you have installed a plugin in your browser

Reply:
Ohk, I see now. I have changed the formatting to JSONLD_FLAT which returns data in one line